src/minikit/hooks/useAuthenticate.ts

```diff
@@ -1,59 +1,145 @@
 import sdk from '@farcaster/frame-sdk';
 import type { SignIn as SignInCore } from '@farcaster/frame-sdk';
 import { useCallback } from 'react';
+import { useMiniKit } from './useMiniKit';
+
+type ParsedSignInMessage = {
+  address: string;
+  'chain-id': string;
+  domain: string;
+  'issued-at': string;
+  nonce: string;
+  resources: string[];
+  uri: string;
+  version: string;
+};
 
 /*
- * Validate the domain of the SWIF message matches the frame's domain set in the .env file
- * @param message - The SWIF message to validate
- * @param domain - The domain of the frame to authenticate against
- * @throws {Error} - Throws an error if the domain or uri is invalid
- * @returns boolean - true if the domain and uri are valid, false otherwise
+ * Parse the sign in message
+ * @param message - The sign in message to parse
+ * @returns {ParsedSignInMessage} - The parsed sign in message
  */
-const validateDomainAndUri = (message: string, domain: string) => {
-  const firstLine = message.split('\n')[0];
-  const domainFromMessage = firstLine.split(
+function parseSignInMessage(message: string): ParsedSignInMessage {
+  const lines = message.split('\n');
+  const [domain] = lines[0].split(
     ' wants you to sign in with your Ethereum account:',
-  )[0];
+  );
+  const address = lines[1];
 
-  const domainUrlObj = new URL(domain);
+  const parsedData: ParsedSignInMessage = {
+    domain,
+    address,
+    'chain-id': '',
+    'issued-at': '',
+    nonce: '',
+    resources: [],
+    uri: '',
+    version: '',
+  };
 
-  // domain in message should match frame's domain
-  if (domainUrlObj.hostname !== domainFromMessage) {
-    throw new Error('Domain mismatch');
-  }
+  return lines.slice(2).reduce((acc, line) => {
+    if (line.includes(': ')) {
+      const [key, value] = line.split(': ');
+      acc[
+        key.replace(/\s/g, '-').toLowerCase() as keyof Omit<
+          ParsedSignInMessage,
+          'resources'
+        >
+      ] = value;
+    } else if (line.startsWith('- ')) {
+      acc.resources = acc.resources.concat(line.slice(2));
```

Comment:
suggestion(non-blocking): Most trivial comment, but could just push since we're already mutating the object